### ISSUE TITLE
Incorrect error message when referencing non-existent type

### ISSUE THREAD

**Issue by davidpcaldwell** [microsoft/TypeScript#41935]
**TypeScript Version:** 4.0.5

**Search Terms:** "no exported member"

**Code**

foo.d.ts
```ts
namespace foo.bar {
	export interface I {
		x: string
	}
}
```
foo.js
```js
//@ts-check

/** @type { foo.bar.I } */
var i;

/** @type { foo.bar.X } */
var x;

/** @type { foo.bar.I.y } */
var y;

/** @param { foo.bar.I.x } p */
function f(p) {}
```

jsconfig.json
```json
{
	"exclude": [
	]
}
```
**Expected behavior:**
Should give errors like:
foo.bar doesn't have an exported type X
foo.bar.I is not a namespace
foo.bar.I is not a namespace

**Actual behavior:**
```
foo.js:6:21 - error TS2694: Namespace 'foo.bar' has no exported member 'X'.

6 /** @type { foo.bar.X } */
                      ~

foo.js:9:21 - error TS2694: Namespace 'foo.bar' has no exported member 'I'.

9 /** @type { foo.bar.I.y } */
                      ~

foo.js:12:22 - error TS2694: Namespace 'foo.bar' has no exported member 'I'.

12 /** @param { foo.bar.I.x } p */                  ~
```
**Playground Link:** Wasn't able to figure out how to create a new playground; I did examine the one for issue #27630 which resembles this one ([link](https://www.typescriptlang.org/play?#code/MYewdgzgLgBAbjAvDALAJgNwFgBQUCeADgKYyzJgCuAtgEbEBO2OYAhtcRIa8KWDAG8AvrlCRYccqky4CJeFKp1GzMdHj9k6Zmw5cepOP2GyipKJphL6TXLs7deZYyJx32DgzACygmLhhAmGIAD0IQBlg1CSRpZiDgsIjYOXNY6xUAoNDwyKsPfScXLMCc5JhohVjtEsTclLMqihobeOykvMqjapkcBLK8+0LDYr72+rJGi3SWzLHSjtghx3MXBNxXXAB6LZgAQWAoSlYAGxgASwgYAHkAaQAuGAByKCeYcBP8GAZiADNGK5QEAwViTEgAGhgtEosEuUOI5zAAHMYJQIMQACYgq6guCnSikAAWjGIADpROB1HAAIyxKDMHb7Q7HE6PAAKDBAJEiXxebwxIE4VhAsFCl1g4DBpBeZhAvx8T3JOEZHK5IHRGMefPeYE+3z+ALIwNBqUh0NhV3oiJRaMx2JB8HxRJJSq6aFi3lJ9O2uwOR1OFyud0eAGFWGAwCLUej8noVs8wG9WDjHScCa7KRIAMzpBm+5mndmc7kEBP8wVXSOikLinVS56pOUKpUq4vqzGh8NV6N8Arxp6J+24p0Z8TwFAe0lgPNM-2smBhiNR22xzxOF6D5MOvFpsk+mCq8IarUWN4fL4-f4MQHG+vhrGg5YGM0wwPw609h8pncEmDEn6jlSACsdLTvufoskWXKMKWG7lkK3ZiuokqpA2sryt4ir7oe7aag2g7nvqV43g6qH3g6T68C+Frvsin5Dqmv7-nuOBdAAbJOFjMOBBbzou3a-IiWL2M8cBYcquw4ceolnrqF4GteRrbk61FvladG2l+pGNMxSqoVAtLIHAM4QYWMAAHJ9l4TyYX+W6RnUyR2hwygMDJLaSW20lPGJOp6pehpAspu6qXC6k2hqDGobppjyFA7rIJ6xk8XOb7BguXbLjGlHSpuKapHpUw5hQJm8Y8llxtZtmEvZwIDFAznEK5ZYeQeXkdhlS6wCuOVllFZiFXFE6JVOpVzp2XU9quwyiXl2kkK1UkdT5BFyURgW3j+pDkY+VlUVCr5hQiGmRVuJo6S6sXmCBRlgRJs6QRZe3StVtWOZEjXNSt4mtmq3lGLJ-kKSRw67iCYAPtNKyhZax0RXaZ31jFeBTBxI1GNx92mXxmWwIJENQ9ZYmLe1eE+YD8nEUpoMEjDtHw1pu2VU4yOoWABmxHApIgFyEBjY9FVri9bw1ZWdWLJ9NjudhpNar5hEBYpQU08QdPhfRiO9azjTswlPiklzPOEHzKUsmlDwNhT61K7epoHTR6uaQxWuXSj8js8VZDc7z-NmYLM02SLb31ZLjCWyTf3La8fmUxt82q-balwxrKYuwBV1WFAw361A3vG7784+dHCvA9TUpq8nTua89f6u79R7LZIVuKyDjHbQT50QonR0flXqc19r7tQDdCh5ybWNlU9zPC3ZYvvQ1WIuVLRc-Z5kdk03MfW63W3g5Ddvmknvenf30+1+nbt8FAaP65IY8YEAA))

**Related Issues:** Possibly issue #27630 as mentioned above

**Comment by RyanCavanaugh** [comment]
This is fixed. With the posted `foo.d.ts`, `foo.js`, and `jsconfig.json`, TypeScript 4.0.5 reports TS2694 for `foo.bar.I.y` and `foo.bar.I.x`, incorrectly saying that `foo.bar` has no exported member `I`.

The adjacent builds are 4.5.0-dev.20210903 (affected) and 4.5.0-dev.20210904 (fixed). Current native TypeScript 7.1.0-dev.20260905.1 also reports TS2713 at those accesses, correctly explaining that `I` is a type rather than a namespace and suggesting `I["y"]` or `I["x"]`. `foo.bar.X` continues to produce the correct TS2694 missing-member diagnostic.

The change was made by merged [#45354](https://github.com/microsoft/TypeScript/pull/45354), commit [`1d51dfa`](https://github.com/microsoft/TypeScript/commit/1d51dfa5504b38f62236acfcc7ebb7506171d9e4).
